algorithm/algorithm.md

```diff
@@ -0,0 +1,147 @@
+# MPM Algorithm
+
+## Main Algorithm
+
+The Material Point Method (MPM) algorithm comprises of 4 major parts.
+
+> **Note** 
+> * MPM stores information at the material points and not at the nodes as it is in Finite Element Method. This includes the application of traction at the particles, not at the nodes.
+> * The code is implementing Modified Update Stress Last (MUSF). The other option is to do Update Stress First (USF) or to do traditional Update Stress last. It is done by moving this block of code to the bottom after the material points velocities and displacements have been updated.
+> * The code computes material points mass in the initialization. Even though the density is updated later, the mass is conserved.
+
+1. Initialization
+
+    1. Read all input files and store necessary information
+
+    1. Compute mass of each material point
```

Comment:
`Mass of each material point is computed based on the density and initial volume. The initial volume is computed based on the material point spacing. The code assumes uniform spacing in all directions. Although the density of material points are updated, the mass is conserved.`